lib/Compilation.js

```diff
@@ -1289,33 +1291,25 @@ class Compilation extends Tapable {
 	applyModuleIds() {
 		const unusedIds = [];
 		let nextFreeModuleId = 0;
-		const usedIds = [];
-		// TODO consider Map when performance has improved https://gist.github.com/sokra/234c077e1299b7369461f1708519c392
-		const usedIdMap = Object.create(null);
+		const usedIds = new Set();
 		if(this.usedModuleIds) {
 			Object.keys(this.usedModuleIds).forEach(key => {
 				const id = this.usedModuleIds[key];
-				if(!usedIdMap[id]) {
-					usedIds.push(id);
-					usedIdMap[id] = true;
-				}
+				usedIds.add(id);
 			});
 		}
 
 		const modules1 = this.modules;
 		for(let indexModule1 = 0; indexModule1 < modules1.length; indexModule1++) {
 			const module1 = modules1[indexModule1];
-			if(module1.id && !usedIdMap[module1.id]) {
-				usedIds.push(module1.id);
-				usedIdMap[module1.id] = true;
+			if(module1.id) {
```

Comment:
@sokra This is filtering module with id `0`. Is it the intended behavior? cc @Kovensky

Reply:
No that is not intended. The module id 0 not not special in any way.

I guess this should filter `null`

Reply:
Please fix it. `module1.id !== null`